src/runtime/virtcontainers/kata_agent.go

```diff
@@ -2259,3 +2267,11 @@ func (k *kataAgent) getAgentMetrics(ctx context.Context, req *grpc.GetMetricsReq
 
 	return resp.(*grpc.Metrics), nil
 }
+
+//defer func(){ err = errorContext("Error from x", err)}()
+func errorContext(con string, err error) error {
```

Comment:
How about use [`pkg/errors`](https://github.com/pkg/errors#adding-context-to-an-error) instead of `errorContext` ?

```golang
_, err := ioutil.ReadAll(r)
if err != nil {
        return errors.Wrap(err, "read failed")
}
```